eclipse-collections-code-generator/src/main/resources/impl/collection/mutable/abstractUnmodifiablePrimitiveCollection.stg

```diff
@@ -353,6 +357,42 @@ public abstract class AbstractUnmodifiable<name>Collection
     {
         return this.collection.injectInto(injectedValue, function);
     }
+
+    @Override
+    public RichIterable\<<name>Iterable> chunk(int size)
+    {
+        if (size \<= 0)
+        {
+            throw new IllegalArgumentException("Size for groups must be positive but was: " + size);
+        }
+        if (this.isEmpty())
+        {
+            return Lists.mutable.empty();
+        }
+
+        MutableList\<<name>Iterable> result = Lists.mutable.empty();
+        if (this.size() \<= size)
+        {
+            Mutable<name>Collection batch = this.collection.newEmpty();
+            batch.addAll(this);
+            result.add(batch);
+        }
+        else
+        {
+            <name>Iterator iterator = this.<type>Iterator();
+            while (iterator.hasNext())
+            {
+                Mutable<name>Collection batch = this.collection.newEmpty();
+                for (int i = 0; i \< size && iterator.hasNext(); i++)
+                {
+                    batch.add(iterator.next());
+                }
+                result.add(batch);
+            }
+        }
+
+        return result;
+    }
```

Comment:
Could this have just been the following instead?

`return this.collection.chunk(size)`